### PR TITLE
replace date[time|multiple] picker by flatpickr

### PR DESCRIPTION
**Pull request by orthagh**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

Replace all old jquery datepickers by [flatpickr](https://flatpickr.js.org).

image

Remove 3 libs by 1 : 

- jquery-ui-datepicker [outdated]
- jquery-ui-multidatespicker [outdated|hacky]
- jquery-ui-timepicker-addon [outdated|bugged]

Fixes /enhancement : 

- small gain in size on page loading
- fix now button issue (internal 19568)
- manage all features in one lib
- use up to date and maintained lib
- lot of clean of unused vars and copypasta (timestep)

Missing detected feature(s) / Cons

- yearrange (showDateField parameter) -> [jquery api](https://api.jqueryui.com/datepicker/#option-yearRange), no usage in glpi code nor in my local plugins (~40), i think we can drop, the feature is not this important.
- display is bit clunky in firefox (select in month  input)
image

### CODE REVIEW

File: inc/html.class.php [hunk 1]
```diff
@@ -2740,118 +2750,90 @@ static function showMassiveActions($options = []) {
...
-      $js .= "}).next('.ui-datepicker-trigger').addClass('pointer');";
-
-      if ($p['multiple']) {
-         // Fix altField date format that has just been filled by multiDatePicker
-         $js .= "normalizeMultiDateAltField($('#hiddendate{$p['rand']}'), '{$format}');";
-      }
+      $values = true === $p['multiple']
+         ? explode(', ', $p['value'])
+         : [$p['value']];
+      $values = implode(', ', array_map('Html::convDate', $values));
```
Comment (cedric-anne) [inline inc/html.class.php:2784]: Conversion is made by JS lib.
```suggestion
```

File: inc/html.class.php [hunk 2]
```diff
@@ -2740,118 +2750,90 @@ static function showMassiveActions($options = []) {
...
+         : "";
+      $clear    = $p['maybeempty'] && $p['canedit']
+         ? "<a data-clear  title='".__s('Clear')."'>
+               <i class='fa fa-times-circle pointer'></i>
+            </a>"
+         : "";
+
+      $output = <<<HTML
+      <div class="no-wrap flatpickr" id="showdate{$p['rand']}">
+         <input type="text" name="{$name}" value="{$values}" size="{$p['size']}"
```
Comment (cedric-anne) [inline inc/html.class.php:2800]:
```suggestion
         <input type="text" name="{$name}" value="{$p['value']}" size="{$p['size']}"
```

File: inc/html.class.php [hunk 3]
```diff
@@ -2990,78 +2959,56 @@ static function showDateTimeField($name, $options = []) {
...
+         : "";
+      $disabled = !$p['canedit']
+         ? " disabled='disabled'"
+         : "";
+      $clear    = $p['maybeempty'] && $p['canedit']
+         ? "<a data-clear  title='".__s('Clear')."'>
+               <i class='fa fa-times-circle pointer'></i>
+            </a>"
+         : "";
+      $p['value'] = trim(self::convDateTime($p['value']));
```
Comment (cedric-anne) [inline inc/html.class.php:2973]: Conversion is made by JS lib.
```suggestion
```

File: inc/html.class.php [hunk 4]
```diff
@@ -2740,118 +2750,90 @@ static function showMassiveActions($options = []) {
...
+HTML;
+
+      $date_format = Toolbox::getDateFormat('js');
+
+      $min_attr = !empty($p['min'])
+         ? "minDate: '".self::convDate($p['min'])."',"
+         : "";
+      $max_attr = !empty($p['max'])
+         ? "maxDate: '".self::convDate($p['max'])."',"
+         : "";
```
Comment (cedric-anne) [inline inc/html.class.php:2816]: I did not tested it, but I think dates should not be converted here (see other comments).

**Comment by cedric-anne** [comment]
`GLPIPlanning` JS functions still uses `datepicker` component.
`normalizeMultiDateAltField` JS function is now useless and must be removed (uses old datepicker lib)

**Comment by orthagh** [comment]
cleaned all convDate + fixed planning

**Comment by orthagh** [comment]
i missed to replace Html::showTimeField, will do next.

File: inc/html.class.php [hunk 5]
```diff
@@ -3051,89 +3051,66 @@ public static function showTimeField($name, $options = []) {
...
-
-      if (!$p['canedit']) {
-         $js .= ",disabled: true";
-      }
-
-      $js .= ",timeFormat: 'HH:mm'";
-      $js .= "}).next('.ui-datepicker-trigger').addClass('pointer');";
-      $js .= "});";
+      $date_format = Toolbox::getDateFormat('js')." H:i:S";
+
```
Comment (cedric-anne) [inline inc/html.class.php:3106]: Dead code
```suggestion
```

**Pull request merged**